src/fread.c

```diff
@@ -2206,12 +2236,31 @@ int freadMain(freadMainArgs _args)
         ch++;
         Field(&fctx);  // stores the string length and offset as <uint,uint> in colNames[i]
         ((lenOff**) fctx.targets)[8]++;
+        if (commentChar) {
+          // skip leading whitespace to detect inline comment marker in header row
+          const char *commentPos = ch;
+          while (commentPos < eof && (*commentPos == ' ' || *commentPos == '\t' || *commentPos == '\0')) commentPos++;
```

Comment:
similarly, this looks like it should be abstracted into a helper to re-use shared logic for easier maintenance

Reply:
we now have the `skip_to_comment_or_nonwhite` helper.